backend/CircleForms/Controllers/V1/PostsController.cs

```diff
@@ -174,6 +174,67 @@ public async Task<IActionResult> Post(PostRequestContract postContract)
         return Unauthorized();
     }
 
+    /// <summary>
+    ///     Update post. (Requires auth)
+    /// </summary>
+    [Authorize]
+    [HttpPatch(ApiEndpoints.PostUpdatePost)]
+    [ProducesResponseType(typeof(PostResponseContract), StatusCodes.Status200OK, "application/json")]
+    public async Task<IActionResult> UpdatePost([FromBody] PostUpdateRequestContract updateContract, string id)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest();
+        }
+
+        var claim = HttpContext.User.Identity?.Name;
+        if (string.IsNullOrEmpty(claim) || !long.TryParse(claim, out _))
+        {
+            return Unauthorized();
+        }
+
+        var post = await _postRepository.Get(id);
+        if (post.AuthorId != claim)
+        {
+            return Unauthorized();
+        }
+
+        _logger.LogInformation("User {Claim} updated the post {Id} with {@Updates}", claim, post.ID, updateContract);
+
+        var questions = post.Questions;
+        var updatedPost = _mapper.Map(updateContract, post);
+        if (updateContract.Questions is not null)
+        {
+            foreach (var updatedPostQuestion in updateContract.Questions)
+            {
```

Comment:
It doesn't seem to handle question removal

Reply:
:/

Reply:
done